src/Processors/Formats/Impl/NpyOutputFormat.cpp

```diff
@@ -0,0 +1,265 @@
+#include <Processors/Formats/Impl/NpyOutputFormat.h>
+
+#include <Core/TypeId.h>
+#include <DataTypes/DataTypeFixedString.h>
+#include <DataTypes/DataTypeArray.h>
+#include <Columns/ColumnsNumber.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnFixedString.h>
+#include <Columns/ColumnArray.h>
+#include <IO/WriteHelpers.h>
+#include <IO/WriteBufferFromString.h>
+#include <Formats/FormatFactory.h>
+
+#include <Common/assert_cast.h>
+
+
+namespace DB
+{
+
+namespace ErrorCodes
+{
+    extern const int TOO_MANY_COLUMNS;
+    extern const int BAD_ARGUMENTS;
+    extern const int ILLEGAL_COLUMN;
+}
+
+namespace
+{
+
+template <typename ColumnType, typename ValueType>
+void writeNumpyNumbers(const ColumnPtr & column, WriteBuffer & buf)
+{
+    const auto * number_column = assert_cast<const ColumnType *>(column.get());
+    for (size_t i = 0; i < number_column->size(); ++i)
+        writeBinaryLittleEndian(ValueType(number_column->getElement(i)), buf);
+}
+
+template <typename ColumnType>
+void writeNumpyStrings(const ColumnPtr & column, size_t length, WriteBuffer & buf)
+{
+    const auto * string_column = assert_cast<const ColumnType *>(column.get());
+    for (size_t i = 0; i < string_column->size(); ++i)
+        buf.write(string_column->getDataAt(i).data, length);
```

Comment:
`length` is the maximum size of all strings in all columns, string in current row can have smaller size, and we should use it's size and use padding with 0 bytes to complete the size to `length`

Reply:
Do you mean `buf.write(string_column->getDataAt(i).data, string_column->getDataAt(i).size);`
I try this, but a read error occurs. 
```
Code: 33. DB::Exception: Received from localhost:9000. 
DB::Exception: Cannot read all data. 
Bytes read: 2. Bytes expected: 4.: (at row 2): (in file/uri /var/lib/clickhouse/user_files/02895_npy_output_format_test_e131dvqu/02895_nested_string.npy): 
While executing NpyRowInputFormat: While executing File. (CANNOT_READ_ALL_DATA)
```

Reply:
I mean:
```c++
buf.write(string_column->getDataAt(i).data, string_column->getDataAt(i).size);
writeChar(0, length -  string_column->getDataAt(i).size, buf);
```

Reply:
Thanks for pointing it out.